### PR TITLE
FIX build failed / Change to use `String#start_with?` instead of `String#starts_with?`.

### PR DESCRIPTION
**Pull request by akiko-pusu**
In Rails edge, deprecate `starts_with?` and `ends_with?` for String core extensions.
Ref https://github.com/rails/rails/pull/39152.

So it seems pull requests are failed since this summer. (#344 and later issues)

Prior to this pull request, I confirmed this commit on my forked repository.
https://github.com/akiko-pusu/offsite_payments/pull/2

### CODE REVIEW

**Comment by akiko-pusu** [comment]
To repository owner or maintainer:

I guess, the Travis CI build for this pull request would have been put on hold due to the limitation of OSS credits.
(Related https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing)

I would appreciate it if someone could confirm this.

**Review by douglas — APPROVED** [review]
Thanks for the contribution, @akiko-pusu! Not sure about the status of Travis CI but we can check.

**Review by nebdil — APPROVED** [review]
LGTM

**Pull request merged**
